test/test_jit.py

```diff
@@ -6295,6 +6295,77 @@ def forward(self, x, hiddens):
             torch.testing.assert_allclose(out, ref)
 
+    @unittest.skipIf(TEST_WITH_UBSAN or not torch.fbgemm_is_cpu_supported(),
+                     'Quantized RNN requires FBGEMM. FBGEMM does not play'
+                     ' well with UBSAN at the moment, so we skip the test if'
+                     ' we are in a UBSAN environment.')
+    def test_gru_quantized(self):
+        d_in, d_hid = 2, 2
+        cell = torch.nn.GRU(d_in, d_hid, 1).float()
+        #
+        vals = [[100, -155],
+                [100, -155],
+                [-155, 100],
+                [-155, 100],
+                [100, -155],
+                [-155, 100],
+                [-155, 100],
+                [100, -155]]
+        vals_bias = np.transpose([1, 2, 3, 4, 5, 6])
+        num_chunks = 3
+        vals = vals[:d_hid * num_chunks]
+        cell.weight_ih_l0 = torch.nn.Parameter(
+            torch.tensor(vals, dtype=torch.float),
+            requires_grad=False)
+        cell.weight_hh_l0 = torch.nn.Parameter(
+            torch.tensor(vals, dtype=torch.float),
+            requires_grad=False)
+        cell.bias_ih_l0 = torch.nn.Parameter(
+            torch.tensor(vals_bias, dtype=torch.float),
+            requires_grad=False)
+        cell.bias_hh_l0 = torch.nn.Parameter(
+            torch.tensor(vals_bias, dtype=torch.float),
+            requires_grad=False)
+        # import pdb
+        # pdb.set_trace()
+        niter = 10
+        x = torch.tensor([[100, -155],
+                          [-155, 100],
+                          [100, -155]], dtype=torch.float).unsqueeze(0).repeat(niter, 1, 1)
+
+        ref = copy.deepcopy(cell)
+        cell = torch.jit.quantized.quantize_rnn_modules(cell)
+
+        output, final_hiddens = ref(x)
+        output_q, final_hiddens_q = cell(x)
+        torch.testing.assert_allclose(output, output_q)
+        torch.testing.assert_allclose(final_hiddens, final_hiddens_q)
+
+        class ScriptWrapper(torch.jit.ScriptModule):
+            def __init__(self, cell):
+                super(ScriptWrapper, self).__init__()
+                self.cell = cell
+
+            @torch.jit.script_method
+            def forward(self, x):
+                # type: (torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]
+                return self.cell(x)
+
+        wrapper = ScriptWrapper(cell)
+        #
+        # # Compare quantize scripted module to unquantized
+        script_out, script_hid = wrapper(x)
+        torch.testing.assert_allclose(script_out, output)
+        torch.testing.assert_allclose(script_hid, final_hiddens)
+
+        # Compare export/import to unquantized
+        # wait for the bug fixed to make the forward_impl be script_method
+        # export_import_wrapper = self.getExportImportCopyWithPacking(wrapper)
+        # ei_out, ei_hid = export_import_wrapper(x)
+        # torch.testing.assert_allclose(ei_out, output)
+        # torch.testing.assert_allclose(ei_hid, final_hiddens)
```

Comment:
Why are these commented?

Reply:
Need to wait for the fix of script bug ("Optional type does not work properly in script_method during tracing").